src/common.rkt

```diff
@@ -101,7 +101,11 @@
 ;; Simple floating-point functions
 
 (define (ordinary-float? x)
-  (and (real? x) (not (or (infinite? x) (nan? x)))))
+  (match x
+    [(? real?)
+     (not (or (infinite? x) (nan? x)))]
+    [(? complex?)
+     (and (ordinary-float? (real-part x)) (ordinary-float? (imag-part x)))]))
```

Comment:
Does it make sense to add support for booleans as well? I don't really know whether it would come up, but it would mean we could think of a notion of `value` which is any intermediate value in Herbie.